### ISSUE TITLE
add ability to plot learning curves

### ISSUE THREAD

**Issue by manjushree2012** [tensorflow/cleverhans#585]
### Visualizing how my deep learning model performed, I need to plot my model's training history in cleverhans. (i.e. epochs vs accuracy) 

Earlier with keras, code like this would do the job. 

```
...  
 ...  
 import matplotlib.pyplot as plt  
   
 ...  
 ...  
 # code for building your model  
 ...  
 ...  
   
 # train your model  
 history = model.fit(train_data, train_labels,  
                     nb_epoch=nb_epoch, batch_size=32)  
   
 print(history.history.keys())  
   
 plt.figure(1)  
   
 # summarize history for accuracy  
   
 plt.subplot(211)  
 plt.plot(history.history['acc'])  
 plt.plot(history.history['val_acc'])  
 plt.title('model accuracy')  
 plt.ylabel('accuracy')  
 plt.xlabel('epoch')  
 plt.legend(['train', 'test'], loc='upper left')  
```

### Now using cleverhans, I am finding accuracy differently, I am wondering how the graph I would like to plot would be possible as code like : 

 ```
history = model.fit(train_data, train_labels,  
                     nb_epoch=nb_epoch, batch_size=32)
```
has been replaced with code like:

```
...........................
#code for building model
................................

# Create TF session and set as Keras backend session
sess = tf.Session()
keras.backend.set_session(sess)

#tensorflow placeholder variables
x = tf.placeholder(tf.float32,shape = (None, X_train_scaled.shape[1]))
y = tf.placeholder(tf.float32,shape = (None, 5))
 
model = mlp_model() 
predictions = model(x)
init = tf.global_variables_initializer()
sess.run(init)

#Train the model
train_params = {
                'nb_epochs': 100,
                'batch_size' : 128,
                'learning_rate' : 0.1,
                'verbose' : 0   
                }

rng = np.random.RandomState([2018, 9, 24])

#train the model, not test
model_train(sess, 
            x, 
            y, 
            predictions, 
            X_train_scaled, 
            Y_train, 
            evaluate=evaluate,
            args = train_params)

```

**Comment by manjushree2012** [comment]
### Thanks for the info.. Also, how can I calculate evaluation metrics other than 'accuracy' like precision, recall etc Because, . 

```
model_train(sess, 
            x, 
            y, 
            predictions, 
            X_train_scaled, 
            Y_train, 
            evaluate=evaluate,
            args = train_params)
```
**returns boolean whereas,** 

`model_eval(sess, x, y, predictions, X_test_scaled, Y_test, args = eval_params)`

**returns a float with accuracy value.**

**Comment by goodfeli** [comment]
That feature doesn't exist yet. If you'd like to add precision / recall metrics for detectors you're welcome to. So far we actually don't have any detector-based functionality that I know of though. I have implemented accuracy / coverage metrics for multiclass classifiers and will send a pull request soon.

**Comment by npapernot** [comment]
Some of this has been implemented by Ian in:

https://github.com/tensorflow/cleverhans/tree/master/cleverhans/plot
https://github.com/tensorflow/cleverhans/tree/master/scripts

I'll close this because we are trying to keep CleverHans to core features needed to benchmark models rather than visualize learning curves.
